fbw-a32nx/src/systems/systems-host/systems/FWC/PseudoFWC.ts

```diff
@@ -1332,6 +1503,194 @@ export class PseudoFWC {
+
+    const voluntaryApDisc =
+      this.autoPilotDisengagedInstantPulse.read() && this.autoPilotInstinctiveDiscPressedInLast1p9Sec.read();
+    this.autoPilotOffVoluntaryEndAfter1p9s.write(voluntaryApDisc, deltaTime);
+    this.autoPilotOffVoluntaryDiscPulse.write(voluntaryApDisc, deltaTime);
+
+    this.autoPilotOffVoluntaryFirstCavalryChargeActive.write(this.autoPilotOffVoluntaryDiscPulse.read(), deltaTime);
+
+    this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.write(
+      this.autoPilotInstinctiveDiscPressedPulse.read() &&
+        (this.autoPilotInstinctiveDiscCountSinceLastFwsCycle > 1 || apDiscPressedInLast1p8SecBeforeThisCycle),
+      deltaTime,
+    );
+
+    this.autoPilotOffVoluntaryMemory.write(
+      this.autoPilotOffVoluntaryDiscPulse.read(),
+      apEngaged ||
+        this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() ||
+        !this.autoPilotOffVoluntaryEndAfter1p9s.read(),
+    );
+
+    const discPbPressedAfterDisconnection =
+      !this.autoPilotDisengagedInstantPulse.read() &&
+      (this.autoPilotInstinctiveDiscPressedPulse.read() || masterWarningButtonLeft || masterWarningButtonRight);
+
+    this.autoPilotOffUnacknowledged.write(
+      this.autoPilotDisengagedInstantPulse.read(),
+      apEngaged || this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() || discPbPressedAfterDisconnection,
+    );
+
+    this.autoPilotOffInvoluntaryMemory.write(
+      !apEngaged && !this.autoPilotOffVoluntaryMemory.read() && this.autoPilotOffUnacknowledged.read(),
+      !this.autoPilotOffUnacknowledged.read(),
+    );
+    this.autoPilotOffInvoluntary.set(this.autoPilotOffInvoluntaryMemory.read());
+    this.autoPilotOffShowMemo.set(this.autoPilotOffVoluntaryMemory.read() || this.autoPilotOffInvoluntaryMemory.read());
+
+    if (this.autoPilotDisengagedInstantPulse.read()) {
+      // Request quiet CRC one time
+      this.requestMasterWarningFromApOff = true;
+      this.soundManager.setVolume(FwsAuralVolume.Attenuated);
+      this.soundManager.enqueueSound('cavalryChargeOnce'); // On the A320, play first cav charge completely no matter what
+    }
+    if (!this.autoPilotOffVoluntaryFirstCavalryChargeActive.read()) {
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.setVolume(FwsAuralVolume.Full);
+    }
+    if (!this.autoPilotOffVoluntaryMemory.read() && !this.autoPilotOffInvoluntaryMemory.read()) {
+      this.requestMasterWarningFromApOff = false;
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.dequeueSound('cavalryChargeCont');
+      this.soundManager.setVolume(FwsAuralVolume.Full);
+    }
+
+    this.autoPilotInstinctiveDiscPressedPulse.write(false, deltaTime);
+
+    // approach capability downgrade. Debounce first, then suppress for a certain amount of time
+    // (to avoid multiple triple clicks, and a delay which is too long)
```

Comment:
On the real plane you can have multiple triple clicks for e.g., when disconnecting both autopilots during approach sometimes you get the cavalry charge followed by two triple clicks in a row, one due to the downgrade from CAT 3 Dual to CAT 3 single and another from CAT 3 single to CAT 1 ( https://youtu.be/pZAI18IRSnE?t=212). On the A380 it jumps straight away from CAT 3 to CAT 1 so it never happens it seems.

Reply:
Are these multiple triple clicks systematic? I believe there's a suppression of around 3 seconds if multiple downgrades happen. Might be some weird timing behavior which we see in the video. https://youtu.be/uPxlp83krBw?si=YfDG7blRXWTWii6C This one and many others have one triple click for a downgrade from C3D to C1